### PR TITLE
Cleanup & consistent variable refs in `unittests`

### PR DESCRIPTION
**Pull request by portante**
Commit message body below, commit summary is PR title:

    Correct local variable usage and `[ ]` vs `[[ ]]`.

    The logger type unit test now checks that the expected `LOGSDIR`
    environment variable is a real directory (avoids whacky stack
    traces for when it isn't).

### CODE REVIEW

**Review by portante — COMMENTED** [review]
Ready for review.

File: server/bin/test-bin/test_logger_type.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 _NAME_ = "pbench-logger-test"
 cfg_name = os.environ["_PBENCH_SERVER_CONFIG"]
 logdir = os.environ["LOGSDIR"]
+assert os.path.isdir(logdir), f"LOGSDIR={logdir}"
```
Comment (webbnh) [inline server/bin/test-bin/test_logger_type.py:13]: That message seems a little cryptic -- it doesn't mention that the indicated directory doesn't actually exist....

Comment (dbutenhof) [inline server/bin/test-bin/test_logger_type.py:13]: Maybe `f"LOGSDIR={logdir!r} isn't a directory"`??

**Review by webbnh — APPROVED** [review]
Looks OK, other than the nit below.

File: server/bin/unittests [hunk 2]
```diff
@@ -61,23 +61,23 @@ function _run_re_unpack {
...
-    rc=$?
-    echo "---- ${_testopt}/lib/crontab/crontab" >> $_testout
-    if [ -s $_testactlog ]; then
-        echo "++++ $(basename $_testactlog) file contents" >> $_testout
-        cat $_testactlog >> $_testout 2>&1
-        echo "---- $(basename $_testactlog) file contents" >> $_testout
+    # Testing the server activate script, which is used for every other unit
+    # test as well is just a matter of emitting the resulting crontab along
+    # side the rest of the environment state provided by the unit test
+    # framework.
```
Comment (dbutenhof) [inline server/bin/unittests:67]: I know you only changed the wrapping, but this sentence isn't particularly helpful.

We're only dumping the generated crontab into the gold file here, and the comment makes this sound rather more involved and suggests we're actually running/testing the script ... which presumably already happened long ago.

**Review by dbutenhof — DISMISSED** [review]

**Review by dbutenhof — APPROVED** [review]

File: server/bin/test-bin/test_logger_type.py (hunk 1 above)
Comment (portante) [reply server/bin/test-bin/test_logger_type.py]: Sorry missed this comment before I posted a change.

File: server/bin/unittests (hunk 2 above)
Comment (portante) [reply server/bin/unittests]: Fixed.

File: server/bin/test-bin/test_logger_type.py (hunk 1 above)
Comment (portante) [reply server/bin/test-bin/test_logger_type.py]: Fixed.

**Review by dbutenhof — APPROVED** [review]

**Pull request merged**